report_account_move_settlement/reports/report_settlement_templates.xml

```diff
@@ -35,8 +35,11 @@
                         style="max-height: 45px;"
                         alt="Logo"
                     />
-                    <t t-if="company.seller_settlement_report_title">
-                        <span t-field="company.seller_settlement_report_title" />
+                    <t t-if="report.report_title">
+                        <span t-field="report.report_title" />
+                    </t>
+                    <t t-elif="company.report_title">
+                        <span t-field="company.report_title" />
                     </t>
```

Comment:
Another idea is to have corresponding compute fields (or just methods) in account.move and just show computed results in the report, to avoid bloating the template with a lot of logic.